hazelcast/src/test/java/com/hazelcast/client/topic/ClientTopicTest.java

```diff
@@ -177,6 +179,39 @@ public void run() {
         });
     }
 
+    @Test
+    public void testNullMessageDecoder() {
```

Comment:
The test name is confusing. I know what you mean but only because I know the background. 
Can you rename it to something like testPublishAllAsync_thenJoin ?

Reply:
Renamed it.